core/tests/test_ingress.py

```diff
@@ -0,0 +1,237 @@
+    else:
+        assert session.query(SensorClass).count() == len(sensor_df.index)
+
+    session_close(session)
+
+def test_import_sensor_location():
+   
+    test_csv = "sensor_location.csv"
+
+    sensor_df = pd.read_csv(os.path.join(CONST_COREDATA_DIR, test_csv))
+    assert sensor_df.empty == False
+
+    # Try to connect to an engine that exists
+    status, log, engine = connect_db(SQL_CONNECTION_STRING, TEST_DB_NAME)
+    assert status, log
+
+    # Creates/Opens a new connection to the db and binds the engine
+    session = session_open(engine)
+
+    # Check if table is empty and bulk inserts if it is
+    first_entry = session.query(SensorLocationClass).first()
+    if first_entry == None:
+        session.bulk_insert_mappings(SensorLocationClass, sensor_df.to_dict(orient="records"))
+        assert session.query(SensorLocationClass).count() == len(sensor_df.index)
+    else:
+        assert session.query(SensorLocationClass).count() == len(sensor_df.index)
+
+    session_close(session)
+
+     # Trying to upload location history data for a sensor that does not exist
+    test_csv = "sensor_location_test_1.csv"
+
+    sensor_df = pd.read_csv(os.path.join(CONST_TEST_DIR_DATA, CONST_SENSOR_LOCATION_TESTS, test_csv))
+    assert sensor_df.empty == False
+    status, log, engine = connect_db(SQL_CONNECTION_STRING, TEST_DB_NAME)
+    assert status, log
+    session = session_open(engine)
+
+    try:
+        session.bulk_insert_mappings(SensorLocationClass, sensor_df.to_dict(orient="records"))
+        result = True
+    except:
+        session.rollback()
+        result = False
+
+    assert result == False
+
+    session_close(session)
+
+    # Trying to upload location history data for a location that does not exist
+    test_csv = "sensor_location_test_2.csv"
+
+    sensor_df = pd.read_csv(os.path.join(CONST_TEST_DIR_DATA, CONST_SENSOR_LOCATION_TESTS, test_csv))
+    assert sensor_df.empty == False
+    status, log, engine = connect_db(SQL_CONNECTION_STRING, TEST_DB_NAME)
+    assert status, log
+    session = session_open(engine)
+
+    try:
+        session.bulk_insert_mappings(SensorLocationClass, sensor_df.to_dict(orient="records"))
+        result = True
+    except:
+        session.rollback()
+        result = False
+
+    assert result == False
+
+    session_close(session)
+
+    # Trying to upload location history data with an empty installation date
+    test_csv = "sensor_location_test_3.csv"
+
+    sensor_df = pd.read_csv(os.path.join(CONST_TEST_DIR_DATA, CONST_SENSOR_LOCATION_TESTS, test_csv))
+    assert sensor_df.empty == False
+    status, log, engine = connect_db(SQL_CONNECTION_STRING, TEST_DB_NAME)
+    assert status, log
+    session = session_open(engine)
+
+    try:
+        session.bulk_insert_mappings(SensorLocationClass, sensor_df.to_dict(orient="records"))
+        result = True
+    except:
+        session.rollback()
+        result = False
+
+    assert result == False
+
+    session_close(session)
```

Comment:
Can we reuse the code from test_populate_db?

Reply:
ongoing conv.

Reply:
I have implemented these changes in another branch.